File: golangLeetcode/1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	resMap := make(map[int]int)
	for i, v := range nums {
		if j, ok := resMap[target-v]; ok {
			return []int{i, j}
		} else {
			resMap[v] = i
		}
	}
	return []int{-1, -1}
}

// @lc code=end

File: swfitLeetcode/1.两数之和.swift
/*
 * @lc app=leetcode.cn id=1 lang=swift
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map = [Int:Int]()
        for (index, val) in nums.enumerated() {
            let t = target - val
            guard let r = map[t] else {
                map[val] = index
                continue
            }
            return [r,index]
        }
        return []
    }
}
// @lc code=end

